openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTrainerTest.java

```diff
@@ -423,7 +440,7 @@ private static ArrayList<List<Double>> fitModelAndGetFirstScoresPerClass(
 
         final LightGBMBinaryClassificationModel model = fit(
```

Comment:
`LightGBMBinaryClassificationModel` is a autoclosable. this means that we should make use of it.

1- `LightGBMBinaryClassificationModel::close()` should be responsible to release all the resources related with the instance.
2- Use `Try...With...Resources` (similar to python With clause) to ensure that resources are released

Example: 

```
try(final LightGBMBinaryClassificationModel model = fit(...)){
   return getClassScores(dataset, model, maxInstancesToScore);
}
// You don't need to explisitly call the close() `Try...With...Resources` will do that for you.
```
 
 We should do that for all Autoclosable resources.

Reply:
Agreed

But perhaps all of these autoclosable try/with clauses should be in the same PR, and this doesn't relate to the FairGBM provider specifically.

Shall we address these on #123 ?

Reply:
actions:
- open new issue with Gandola's comment
- add TODO in code referring to ticket

Reply:
@AndreFCruz : Add this issue link with a TODO on the code in the line before: https://github.com/feedzai/feedzai-openml-java/issues/130